App crashing when a user is taking rdt image

Check logs below

06-03 19:29:50.961   406 18922 W QCamera2HWI: preview_stream_cb_routine : returned!
06-03 19:29:50.972   433 18919 E         : af_process_parse_stats: Invalid FV data!
06-03 19:29:50.972   433 18919 E         : af_process_1: Error processing AF stats!
06-03 19:29:50.990 17638 18789 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #6
06-03 19:29:50.990 17638 18789 E AndroidRuntime: Process: io.ona.rdt.covid.id, PID: 17638
06-03 19:29:50.990 17638 18789 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at android.os.AsyncTask$3.done(AsyncTask.java:325)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:762)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: Caused by: java.lang.NoSuchMethodError: No static method encodeHexString([B)Ljava/lang/String; in class Lorg/apache/commons/codec/binary/Hex; or its super classes (declaration of 'org.apache.commons.codec.binary.Hex' appears in /system/framework/org.apache.http.legacy.boot.jar)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at org.apache.commons.codec.digest.DigestUtils.md5Hex(DigestUtils.java:310)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at io.ona.rdt.util.RDTJsonFormUtils.getFileMD5Hash(RDTJsonFormUtils.java:199)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at io.ona.rdt.util.RDTJsonFormUtils.saveImage(RDTJsonFormUtils.java:119)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at io.ona.rdt.util.RDTJsonFormUtils.access$000(RDTJsonFormUtils.java:73)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at io.ona.rdt.util.RDTJsonFormUtils$1SaveImageTask.doInBackground(RDTJsonFormUtils.java:92)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at io.ona.rdt.util.RDTJsonFormUtils$1SaveImageTask.doInBackground(RDTJsonFormUtils.java:88)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at android.os.AsyncTask$2.call(AsyncTask.java:305)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
06-03 19:29:50.990 17638 18789 E AndroidRuntime: 	... 4 more
06-03 19:29:50.994   636 27153 D Debug   : !@DumpState : SHIP
06-03 19:29:50.994   636 27153 D Debug   : !@DumpState : debug level:0x4f4c
06-03 19:29:50.995   636 27153 D Debug   : !@Dumpstate : Finally, system will skip dumpstate

--- a/rdt-open-reader/src/main/java/io/ona/rdt/util/RDTJsonFormUtils.java
+++ b/rdt-open-reader/src/main/java/io/ona/rdt/util/RDTJsonFormUtils.java
@@ -9,6 +9,7 @@ import android.os.AsyncTask;
 import androidx.core.util.Pair;
 import android.widget.Toast;
 
+import org.apache.commons.codec.binary.Hex;
 import org.apache.commons.codec.digest.DigestUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.json.JSONArray;
@@ -196,7 +197,7 @@ public class RDTJsonFormUtils {
         InputStream inputStream = null;
         try {
             inputStream = new FileInputStream(absoluteFilePath);
-            md5Hash = DigestUtils.md5Hex(inputStream);
+            md5Hash = new String(Hex.encodeHex(DigestUtils.md5(inputStream)));
         } catch (IOException e) {
             Timber.e(e);
         } finally {

